agent/agent.go

```diff
@@ -234,14 +241,29 @@ func (r *Agent) setupSink(ctx context.Context, sr recipe.SinkRecipe, stream *str
 		return
 	}
 
-	stream.subscribe(func(records []models.Record) (err error) {
-		err = sink.Sink(ctx, records)
+	retryNotification := func(e error, d time.Duration) {
```

Comment:
@StewartJingga I think we can avoid this abstraction right. Since this retrier is very specific to the agent anyway. and we do not plan to expose it outside. Wdyt?

Reply:
sorry which abstraction? the `retryNotification`?
oh was just moving it from creating it again and again as anonymous func to a reusable functions, also im able to name it via variable for clarity. But im not against changing it back to anonymous func if you want.

btw it is still inside the agent.go code so nothing is exposed there, sink does need to know about it, this is just how agent.go handling sink.

Reply:
Ok. Lets skip for now.